projects/storefrontapp-e2e-cypress/cypress/e2e/regression/cms/cms-navigation.core-e2e.cy.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { isolateTests } from '../../../support/utils/test-isolation';
+
+context('Cms navigation', { testIsolation: false }, () => {
+  isolateTests();
+  it('should not go into infinite loop of loading cms components data, when some details of some component are missing', () => {
+    /**
+     * Circuit breaker, just in case if the infinite loop would happen. However, in a successful test pass, infinite loop should not happen.
+     */
+    let circuitBreakerCounter = 20;
+    function shouldSimulateMissingData(): boolean {
+      return circuitBreakerCounter-- >= 0;
+    }
+
+    // We want to intercept the HTTP response from `/cms/components` endpoint and simulate missing details of one random CMS component.
+    // This endpoint is expected to be called a few times, but not infinite number of times.
+    // Just in case we would go into infinite loop, we have circuit breaker.
+    cy.intercept('GET', `**/cms/components*`, (req) => {
+      req.reply((res) => {
+        if (shouldSimulateMissingData()) {
+          res.body.component.pop();
+        }
+        return res;
+      });
+    }).as('getComponents');
+
+    cy.visit('/');
+    cy.wait('@getComponents');
+    cy.get('cx-category-navigation cx-generic-link').should('exist');
+
+    // `/cms/components` endpoint is called for each different Navigation components (e.g. header and footer)
+    const EXPECTED_MAX_COUNT_LOADING_CMS_COMPONENTS = 3;
+
+    // Our heuristic to assert that there is no infinite loop of calling `/cms/components` endpoint
+    // is to check whether this endpoint was called not too many times during 1st second after the page is rendered.
+    cy.wait(1_000);
+    cy.get('@getComponents.all', { timeout: 0 }).should(
```

Comment:
Pasting here my explanation from Slack, just for reference:

> 
> I had problem to choose the right expected amount of calls to /components
> 
> 1. it's called once per every navigation - we have 2 : top nav and footer nav
> 2. it's called once per a set of missing components in those navigations - according to my mocked response I deliberately remove 1 component from the response: it likely happens both for the call for top nav and call for bottom nav
> 
> therefore there are 4 requests
> 
> I don't know why I wrote 3
> 
> alternatively I could change my test to deliberately remove 1 component of a specific uid, e.g. CanonBrandCategoryLink
> 
> but then the tests would depend hardly on sample data
> 
> currently it depends not that hard on sample data - however it depends on the fact that the CMS structure returned from /pages endpoint cotains only 2 navigations
> 
> if there where 3 navigation menus (top, footer, and something else), then there would be 6 calls to /components in total (3 normal ones, and additional 3 with attempt to load missing components)
> 
> I was scratching my head what would be a good expected number
> 
> I could go for:
> - wait 5-10 seconds
> - and the number of calls should be less than 20 (with infinite loop it is more than 100 by that time)
> 
> bud decided to go:
> - wait 1 second (because it's not that much... we can afford it)
> - and the number of calls should be exact  as expected  (with infinite loop, during this 1st second there can happen only a few calls to /components)